homework-submission/main.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+function AddingForm({ addTodoItem }) {
+  return (
+    <div className='form'>
+      <form onSubmit={addTodoItem} className='Add'>
+        <input type='text' name='description' placeholder='Description' className='AddingList' />
+        <input type='date' name='deadline' placeholder='Deadline' />
+        <input type='submit' name='add' value='Add' className='submit' />
+      </form>
+    </div>
+  );
+}
+
+class TodoList extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      dynamicList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+    };
+  }
+
+  addItem = element => {
+    element.preventDefault();
+    const newTodo = {
+      id: this.state.dynamicList.length + 1,
```

Comment:
This is a solution of making a new key,
perhaps using a random thing would make a better uniquely use of ids, and therefore a unique key for each item in the list, and easier to recognize for react.

this would be a reason for the following message and causing some bugs in the future!
```red
Warning: Encountered two children with the same key, `1`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```

you could replace this value with the following 
```javascript
Math.random().toString(36).substr(2, 10)
```
this will make sure it will be as unique as possible, or you could search google to find this way of making a unique string of letters:
 - one of the following solution would fit:
  https://stackoverflow.com/questions/6860853/generate-random-string-for-div-id 
 - or this:
  https://www.w3resource.com/javascript-exercises/javascript-math-exercise-23.php
 - or even installing this library would make it happen 😃
  _it's just an example you don't have to install anything here!_
  https://www.npmjs.com/package/uuid

And I like simplest solutions better.